src/metrics/mongoose_metrics_mam_hooks.erl

```diff
@@ -16,80 +16,77 @@
-
--type metrics_notify_return() :: mongoose_metrics_hooks:metrics_notify_return().
-
 %%-------------------
 %% Implementation
 %%-------------------
 
 %% @doc Here will be declared which hooks should be registered when mod_mam_pm is enabled.
--spec get_mam_hooks(_) -> [ejabberd_hooks:hook(), ...].
+-spec get_mam_hooks(_) -> gen_hook:hook_list().
 get_mam_hooks(Host) ->
     [
-        {mam_set_prefs, Host, ?MODULE, mam_set_prefs, 50},
-        {mam_get_prefs, Host, ?MODULE, mam_get_prefs, 50},
-        {mam_archive_message, Host, ?MODULE, mam_archive_message, 50},
-        {mam_remove_archive, Host, ?MODULE, mam_remove_archive, 50},
-        {mam_lookup_messages, Host, ?MODULE, mam_lookup_messages, 100},
-        {mam_flush_messages, Host, ?MODULE, mam_flush_messages, 50}
+        {mam_set_prefs, Host, fun ?MODULE:mam_set_prefs/3, #{}, 50},
+        {mam_get_prefs, Host, fun ?MODULE:mam_get_prefs/3, #{}, 50},
+        {mam_archive_message, Host, fun ?MODULE:mam_archive_message/3, #{}, 50},
+        {mam_remove_archive, Host, fun ?MODULE:mam_remove_archive/3, #{}, 50},
+        {mam_lookup_messages, Host, fun ?MODULE:mam_lookup_messages/3, #{}, 100},
+        {mam_flush_messages, Host, fun ?MODULE:mam_flush_messages/3, #{}, 50}
     ].
 
 %% @doc Here will be declared which hooks should be registered when mod_mam_muc is enabled.
--spec get_mam_muc_hooks(_) -> [ejabberd_hooks:hook(), ...].
+-spec get_mam_muc_hooks(_) -> gen_hook:hook_list().
 get_mam_muc_hooks(Host) ->
     [
-        {mam_muc_set_prefs, Host, ?MODULE, mam_muc_set_prefs, 50},
-        {mam_muc_get_prefs, Host, ?MODULE, mam_muc_get_prefs, 50},
-        {mam_muc_archive_message, Host, ?MODULE, mam_muc_archive_message, 50},
-        {mam_muc_remove_archive, Host, ?MODULE, mam_muc_remove_archive, 50},
-        {mam_muc_lookup_messages, Host, ?MODULE, mam_muc_lookup_messages, 100},
-        {mam_muc_flush_messages, Host, ?MODULE, mam_muc_flush_messages, 50}
+        {mam_muc_set_prefs, Host, fun ?MODULE:mam_muc_set_prefs/3, #{}, 50},
+        {mam_muc_get_prefs, Host, fun ?MODULE:mam_muc_get_prefs/3, #{}, 50},
+        {mam_muc_archive_message, Host, fun ?MODULE:mam_muc_archive_message/3, #{}, 50},
+        {mam_muc_remove_archive, Host, fun ?MODULE:mam_muc_remove_archive/3, #{}, 50},
+        {mam_muc_lookup_messages, Host, fun ?MODULE:mam_muc_lookup_messages/3, #{}, 100},
+        {mam_muc_flush_messages, Host, fun ?MODULE:mam_muc_flush_messages/3, #{}, 50}
     ].
 
--spec mam_get_prefs(Result :: any(),
-                    Host :: jid:server(),
-                    _ArcID :: mod_mam:archive_id(),
-                    _ArcJID :: jid:jid()) -> any().
-mam_get_prefs(Result, Host, _ArcID, _ArcJID) ->
+-spec mam_get_prefs(Acc, Params, Extra) -> {ok, Acc} when
+    Acc :: any(),
+    Params :: map(),
+    Extra :: #{host_type := mongooseim:host_type()}.
+mam_get_prefs(Acc, _, #{host_type := Host}) ->
     mongoose_metrics:update(Host, modMamPrefsGets, 1),
```

Comment:
We might rename these variables to `HostType`, it is what they actually are but were just not updated when we introduced host types 🙂

Reply:
done 👍